### PR TITLE
Apic 461/get lexical info

### PR DESCRIPTION
**Pull request by carowright**
New method to get lexical information.

### CODE REVIEW

File: src/AmfHelperMixin.js [hunk 1]
```diff
@@ -1222,4 +1222,33 @@ export const AmfHelperMixin = (base) => class extends base {
       }
     });
   }
+
+  /**
+   * Get first appearance in lexical information
+   * @param {Object} value
+   * @return {String|undefined} order.
+   */
+  _getLexicalOrder(value) {
```
Comment (TwoPlusTwoOne) [inline src/AmfHelperMixin.js:1231]: Not to be picky, but just so it'll be easier to understand for future reference, can we rename it to something like `_computeLexicalStart`? And then see my comment below

File: src/AmfHelperMixin.js [hunk 2]
```diff
@@ -1222,4 +1222,33 @@ export const AmfHelperMixin = (base) => class extends base {
...
+    const lKey = this._getAmfKey(
+      this.ns.aml.vocabularies.docSourceMaps.lexical
+    );
+    const lexical = sources[lKey];
+
+    if (!lexical || lexical.length === 0) {
+      return undefined;
+    }
+
+    const firstPosition = lexical[0];
```
Comment (TwoPlusTwoOne) [inline src/AmfHelperMixin.js:1249]: I thought about it, and maybe we should always look for the lowest start value. I know that for what we need it for, it's exactly the same, but it's best if we maintain as much consistency as possible

**Comment by jarrodek** [comment]
Guys, thanks for the PR. I need to know a little bit more about this change (and the related change in the navigation panel). 
1) Does this function is relevant to any other component? Will we used anywhere else? If not then this should be added to the navigation component.
2) What is the issue about? I can see an issue with the ordering of endpoints. The lexical ordering (as you call it) will interfere with building the tree view. In OAS and RAML you can add an endpoint at any place of the spec but this endpoint should be still rendered in the tree with the rest of the endpoints (when it's a sub-endpoint). How this affect this rendering?

**Comment by TwoPlusTwoOne** [comment]
@jarrodek when using queries with the AMF Service, the order of _all_ nodes is not guaranteed. This is most visible with the navigation, but it is also happening in places like headers, query parameters, methods, and more. The idea is that we add this sorting to different parts of the console as needed

**Comment by jarrodek** [comment]
@TwoPlusTwoOne this looks like a query service problem to solve and not ours. The only task of the API Console is to render the model AS IS. If there's missing information in the model then this is a problem for the query service. 
Think of it this way. This fixes very specific problem that is located in Exchange when using query service. The service does not return the ordered list of endpoints. However, query service is basically a graph search. They can do this on the server side much faster then your computations in the browser. 
What do you mean by this is also happening with headers, quest params, methods, etc. How the information is presented for these parts of the model? I understand that there might be a lot of endpoints in an API but you have pretty finite number of methods you can put into an endpoint. Why do you need pagination here? The same for headers or query parameters.
I think we are approaching this problem from the wrong side. The query service should return an information that the console should render without some crazy hacks involved on our side. The effect of this change will be us trying to to do additional data processing in the client side which is not future proof and with any change this adds us a lot of work. I've been doing this for long enough to know that this will cause problems in the future. We should instead contact the query service team to them to return the already processed information on their side. They operate on the entire model when generating the JSON+LD and can return the correct ordering. On the client side, with a limited information, we can only guess. I really don't like this idea.

**Pull request closed without merging**